src/rpc/zmq_pub.cpp

```diff
@@ -143,19 +163,27 @@ namespace
     dest.EndObject();
   }
 
-  void toJsonValue(rapidjson::Writer<epee::byte_stream>& dest, const minimal_txpool self)
+  void toJsonValue(rapidjson::Writer<epee::byte_stream>& dest, const miner_data& self)
   {
-    crypto::hash id{};
-    std::size_t blob_size = 0;
-    if (!get_transaction_hash(self.tx, id, blob_size))
-    {
-      MERROR("ZMQ/Pub failure: get_transaction_hash");
-      return;
-    }
+    dest.StartObject();
+    INSERT_INTO_JSON_OBJECT(dest, major_version, self.major_version);
+    INSERT_INTO_JSON_OBJECT(dest, height, self.height);
+    INSERT_INTO_JSON_OBJECT(dest, prev_id, self.prev_id);
+    INSERT_INTO_JSON_OBJECT(dest, seed_hash, self.seed_hash);
```

Comment:
Thoughts on treating `seed_hash` as optional if `major_version >= RX_BLOCK_VERSION` (here & in other parts of the code too) just in case the endpoint is being used for earlier blocks too?

Reply:
It's always used for the current blockchain height, not for any earlier blocks. Even if it's used for major_version before that, seed_hash will just be all 0, but it only matters for testnet experiments (mining older blocks for example).

Reply:
Sounds fine to me :)